modules/tinymce/src/plugins/table/main/ts/ui/Buttons.ts

```diff
@@ -165,6 +168,56 @@ const addButtons = (editor: Editor, selectionTargets: SelectionTargets, clipboar
     icon: 'table'
   });
 
+  const tableClassList = getTableClassList(editor);
+
+  if (tableClassList.length === 0) {
+    // eslint-disable-next-line no-console
+    console.error('Missing table class list');
+  } else {
+    editor.ui.registry.addMenuButton('tableclass', {
+      icon: 'table-classes',
+      tooltip: 'Table styles',
+      fetch: (callback) => {
+        callback(Arr.map(tableClassList, (value) => {
```

Comment:
This logic is all duplicated below with the exception of the class list, command name and format name. Please pull this out into a helper function to generate the `fetch` callback and just pass those three things in.

- [ ] Remove duplication